TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -715,6 +717,102 @@ public class JunctionRestrictionsManager
+
+            ushort nearestClockwiseSegId = 0;
+            int clockwiseDistance = 0;
+            foreach (var otherSegmentId in ExtNodeManager.Instance.GetNodeSegmentIds(nodeId, ClockDirection.Clockwise, segmentId)) {
+                if (otherSegmentId.ToSegment().Info.m_hasPedestrianLanes) {
+                    nearestClockwiseSegId = otherSegmentId;
+                    break;
+                }
+                ++clockwiseDistance;
+            }
+
+            if (nearestClockwiseSegId == 0) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"no pedestrian lanes on node {nodeId}"
+                                + $"returning false");
+                }
+
+                return false;
+            }
+
+            if (includeAllPossible) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"pedestrian lanes found on segment {nearestClockwiseSegId}"
+                                + $"returning true");
+                }
+
+                return true;
+            }
+
+            ushort nearestCounterClockwiseSegId = 0;
+            int counterClockwiseDistance = 0;
+            foreach (var otherSegmentId in ExtNodeManager.Instance.GetNodeSegmentIds(nodeId, ClockDirection.CounterClockwise, segmentId)) {
+                if (otherSegmentId.ToSegment().Info.m_hasPedestrianLanes) {
+                    nearestCounterClockwiseSegId = otherSegmentId;
+                    break;
+                }
+                ++counterClockwiseDistance;
+            }
+
+            if (nearestCounterClockwiseSegId == 0 || nearestClockwiseSegId == nearestCounterClockwiseSegId) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"no pedestrian lanes on segment {segmentId}"
+                                + $"only one segment on node {nodeId} has pedestrian lanes"
+                                + $"returning false");
+                }
+
+                return false;
+            }
+
+            int distanceThisSide = clockwiseDistance + counterClockwiseDistance - 1;
+            int distanceOtherSide = 0;
+            foreach (var otherSegmentId in ExtNodeManager.Instance.GetNodeSegmentIds(nodeId, ClockDirection.Clockwise, nearestClockwiseSegId)) {
+                ++distanceOtherSide;
+                if (otherSegmentId == nearestCounterClockwiseSegId)
+                    break;
+            }
+
+            bool result = distanceThisSide < distanceOtherSide;
```

Comment:
what if its equal?

Reply:
Also this function does more than the flowchart in the description explains. it checks distance from each side.

Reply:
> what if its equal? which side get zebra crossings?

Each segment is evaluated separately, and this logic is only used if the segment doesn't have pedestrian lanes. Since the segments with pedestrian lanes are always counted in the opposite distance, there is no scenario where segments on opposite sides will both fail this test. For example...

In the first image, we're deciding whether to allow crossings on the segment with the red rectangle. The two distances are equal, so they will not be allowed.
![image](https://user-images.githubusercontent.com/95266227/167151107-9e5b5d5d-51c9-42a7-9176-ce98f5f1284e.png)

In the second image, we're deciding whether to allow crossings on the segment with the green rectangle. But here the opposite distance is longer because it includes all segments with pedestrian lanes, so the crossing will be allowed.
![image](https://user-images.githubusercontent.com/95266227/167152195-6c19160c-4df2-4579-9493-8cd3ba686510.png)

Reply:
> Also this function does more than the flowchart in the description explains. it checks distance from each side too.

It is implied by:

> Does any shortest path go through this segment?

You can't know the shortest path unless you compare it to other potential paths. In this case, the paths are through this segment vs. around the other side.

Reply:
got it thanks :)